bedrock/firefox/templatetags/helpers.py

```diff
@@ -253,13 +254,37 @@ def firefox_url(platform, page, channel=None):
     if channel == 'esr':
         channel = 'organizations'
 
-    if channel:
-        kwargs['channel'] = channel
-    if platform != 'desktop':
-        kwargs['platform'] = platform
+    # There is now only one /all page URL - issue 8096
+    if page == 'all':
+        if platform == 'desktop':
+            if channel == 'beta':
+                anchor = 'product-desktop-beta'
+            elif channel == 'developer':
+                anchor = 'product-desktop-developer'
+            elif channel == 'nightly':
+                anchor = 'product-desktop-nightly'
+            elif channel == 'organizations':
+                anchor = 'product-desktop-esr'
+            else:
+                anchor = 'product-desktop-release'
+        elif platform == 'android':
+            if channel == 'beta':
+                anchor = 'product-android-beta'
+            elif channel == 'nightly':
+                anchor = 'product-android-nightly'
+            else:
+                anchor = 'product-android-release'
+    else:
+        if channel:
+            kwargs['channel'] = channel
+        if platform != 'desktop':
+            kwargs['platform'] = platform
 
     # Firefox for Android and iOS have the system requirements page on SUMO
     if platform in ['android', 'ios'] and page == 'sysreq':
         return settings.FIREFOX_MOBILE_SYSREQ_URL
 
+    if anchor:
+        return reverse('firefox.%s' % page, kwargs=kwargs) + '#' + anchor
+
     return reverse('firefox.%s' % page, kwargs=kwargs)
```

Comment:
Instead of duplicating the reverse call I would do something more like:

```python
anchor = '#' + anchor if anchor else ''
return reverse('firefox.%s' % page, kwargs=kwargs) + anchor
```

What you have is fine, but that's a bit more DRY. And one more note just as an FYI, since we now run on Python 3.7 we can use [f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings). So this reverse could be just:

```python
reverse(f'firefox.{page}', kwargs=kwargs)
```